packages/compass-connections-navigation/src/connections-navigation-tree.tsx

```diff
@@ -388,17 +400,43 @@ const ConnectionsNavigationTree: React.FunctionComponent<
     'enableNewMultipleConnectionSystem'
   );
 
+  const onConnectionExpandRef = useRef(onConnectionExpand);
+  useEffect(() => {
+    onConnectionExpandRef.current = onConnectionExpand;
+  }, [onConnectionExpand]);
+
+  const onDatabaseExpandRef = useRef(onDatabaseExpand);
+  useEffect(() => {
+    onDatabaseExpandRef.current = onDatabaseExpand;
+  }, [onDatabaseExpand]);
```

Comment:
I think we can simplify this as well, probably something like this?
We won't need effect because having it or not having it will have the same result (you modifying the current of the ref). But regardless it remains the same for the consumer because the consumer is dependent on only the ref and can be sure that at any point in time it will refer to the up-to-date version of current.

```suggestion
  const onConnectionExpandRef = useRef();
  onConnectionExpandRef.current = onConnectionExpand;

  const onDatabaseExpandRef = useRef();
  onDatabaseExpandRef.current = onDatabaseExpand;
```